### PR TITLE
Removes the need for `ComponentInitialize`

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
Removes the `ComponentInitialize()` proc and the need for it. All `AddComponent()`s are now down in `Initialize()`

## Why It's Good For The Game
Saves a bit of initialization time due to avoiding proc call overhead. 

## Testing
Compiled, ran server, had no runtimes. Took a look at all objects which were using `ComponentInitialize`. Load order seems to be unaffected, as objects don't have dependency on having a component loaded or not.

## Changelog
Backend only.

### CODE REVIEW

File: code/game/objects/effects/decals/decal.dm [hunk 1]
```diff
@@ -4,12 +4,10 @@
...
-
-/obj/effect/turf_decal/ComponentInitialize()
-	. = ..()
+	..()
+	. = INITIALIZE_HINT_QDEL
 	var/turf/T = loc
 	if(!istype(T)) //you know this will happen somehow
-		CRASH("Turf decal initialized in an object/nullspace")
+		stack_trace("Turf decal initialized in an object/nullspace")
+		return
```
Comment (AffectedArc07) [inline code/game/objects/effects/decals/decal.dm:12]: Why convert CRASH() to stack_trace() return?

File: code/game/machinery/flasher.dm [hunk 2]
```diff
@@ -29,7 +29,7 @@
 	base_state = "pflash"
 	density = TRUE
 
-/obj/machinery/flasher/portable/ComponentInitialize()
+/obj/machinery/flasher/portable/Initialize()
```
Comment (AffectedArc07) [inline code/game/machinery/flasher.dm:32]: Could you make these have the mapload argument just for consistency? 

I know the var isn't used anywhere, but readability would be nice.

File: code/game/objects/effects/decals/decal.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/game/objects/effects/decals/decal.dm:12]: So it can still return `INITIALIZE_HINT_QDEL`.

File: code/game/machinery/flasher.dm (hunk 2 above)
Comment (SteelSlayer) [reply code/game/machinery/flasher.dm:32]: Yeah I can do do that.

File: code/game/objects/effects/decals/decal.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/game/objects/effects/decals/decal.dm:12]: `CRASH()` still returns the `.` value
image

Comment (SteelSlayer) [reply code/game/objects/effects/decals/decal.dm:12]: Neat, didn't know that. Changing it back.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**